graph/src/components/graphql.rs

```diff
@@ -43,6 +43,14 @@ pub trait GraphQlRunner: Send + Sync + 'static {
     ) -> Result<SubscriptionResult, SubscriptionError>;
 
     fn load_manager(&self) -> Arc<LoadManager>;
+
+    fn metrics(&self) -> Arc<dyn GraphQLMetrics>;
+}
+
+pub trait GraphQLMetrics: Send + Sync + 'static {
+    fn observe_query_execution(&self, duration: Duration, results: &QueryResults) -> ();
+    fn observe_query_parsing(&self, duration: Duration, results: &QueryResults) -> ();
+    fn observe_query_validation(&self, duration: Duration, id: &DeploymentHash) -> ();
```

Comment:
The `-> ()` can be omitted.

```suggestion
pub trait GraphQLMetrics: Send + Sync + 'static {
    fn observe_query_execution(&self, duration: Duration, results: &QueryResults);
    fn observe_query_parsing(&self, duration: Duration, results: &QueryResults);
    fn observe_query_validation(&self, duration: Duration, id: &DeploymentHash);
```

Reply:
Ok, did